### ISSUE TITLE
Improve forecast skill for pure ERA5 training

### ISSUE THREAD

**Issue by clessig** [ecmwf/WeatherGenerator#126]
### Is your feature request related to a problem? Please describe.

The current WeatherGenerator has ok skill for training with just ERA5. The forecast skill should be improved to reach roughly Pangu-Weather skill.

### Describe the solution you'd like

A systematic ablation study should be performed to optimize the performance of the WeatherGenerator for ERA5. Parameters that should be considered need to be discussed in more detail.

CC @mlangguth89 @kctezcan

**Comment by mlangguth89** [comment]
Initial experiments in an auto-encoder setting has been conducted, showing that increasing the embedding dimensions in the embedding network and the local assimilation engine can improve the performance of the model. 
However, the overall set-up was found to bee too simplistic for actually be useful for pretraining. In particular, the model has to learn the identity function in ths setting (only ERA5 in- and output, no forecasting, no masked modeling) without any real bottleneck in the data processing. Consequently, the achieved loss has been very low with test loss values of O(10^-3). 

Later experiments now focus on training the model for one-step forecasting. However, the set-up of the model is not yet established.
In a first trial, the model has been trained on forecasting, but with an deactivated forecasting engine (`fe_num_block = 0`). The attained results are rather weak with a test loss in the order of 10^-1. This translates, for instance, to an RMSE of 1.45 K of the 2m temperature for 6h forecasting. Furthermore, it has been found that the 'forecast' for 0h are actually even a bit worse, with some signature of grid artefacts in the output (see `10u` or `q_850`). 
Conceptionally, the above model set-up is also questionable since the atmospheric state at 0h and 6h lead time is sampled from the same latent space. 

Furthermore, experiments question the role of the local assimilation engine, as well as the forecasting engine. 
Below, the loss curves for three experiments are shown that has been run on two nodes of Juwel Booster (=8 A100 GPUs):

![Image](https://github.com/user-attachments/assets/1133cf96-70a8-47ac-bebc-9f276951206f)

While the experiment without a forecasting engine

**Comment by mlangguth89** [comment]
Numerous experiments with varying the following hyperparameters have been performed for the one-step forecast pretraining (no forecast engine, but `forecast_offset: 1`:
- `embed_dim`
- `ae_local_dim_embed`
- `ae_global_dim_embed`
- 'ae_global_num_blocks'

The set of experiments is summarized in the following table:
| exp no | embed \_dim | ae\_local \_dim\_embed  | ae\_global \_dim\_embed | ae\_global \_num\_blocks |assignee/status | train / val IDs |
| --- | --- | --- | --- | --- | --- | --- |
| 0 | default | default | default | default | Michael | qdsfqjr5/c1klxn40 | 
| 1 | default | 2048 | default | default | Michael | gbl3efpo/diw4dn71 | 
| 2 | 512 | default | default | default | Michael | svcbnjku/pizdoger | 
| 3 | 512 | 2048 | default | default | Michael | r58xdngp/kf1sd47t |
| 4 | default | default | 1024 | default |Sindhu | k96gmzhl/ddhyljk4 |
| 5 | 512 | default | 1024 | default | Sindhu | muso9wvz/zi3n94eq |
| 6 | 512 | 2048 | 1024 | default | Ankit | cczsmxl8/ez9e2ws4 |
| 7 | default | default | default | 10 | Ankit | vq7gyc3r/gpil9wb2 |
| 8 | 512 | default | default | 10 | Ankit | jxeafbtg/cy47oi2n |
| **9** | 512 | 2048 | default | 10 | Asma | gaset6wv/dj91346f | 
| 10 | 512 | 2048 | 1024 | 10 | Asma | diqof63l/xp10rzt2 | 

The results in terms of an RMSE-based scorcard are:

![Image](https://github.com/user-attachments/assets/93970255-d438-4ed4-a771-345f2db37983)

The headline experiment No. 9 with increased `embed_dim`, `ae_local_dim_embed` and `ae_gloabl_num_blocks` yields an improvement of 10% over the default configuration in terms of the RMSE for 2t, 10u, 10v, z_850, t_850, u_850, v_850, z_500, t_500, u_500, v_500. 
Further details can be found at the end of the following hedgedoc: https://gitlab.jsc.fz-juelich.de/hedgedoc/xuYwPs8qRNa5PRvoK_oTyg?view

**Comment by SavvasMel** [comment]
> [@mlangguth89](https://github.com/mlangguth89) what are the conclusions from this work so far? i am not sure what else needs to be done for this specific ticket. Are there concrete follow-ups?

Maybe I can try to give a bit of update here: we have moved to pretraining the models with a masking strategy with the configurations tried in this work above. Of course, it should be noted here, that the configurations above were tried under an auto-encoder setting (no masking) but still I believe they can be useful also for masking training. 

For example below I am using the "best" configuration tried above (number 9), and I have got a quite good pretrained model on 0h forecast. If you choose the "default" configuration model you do not get such a good performance. I should also add here, that we have models that even better than the one below but those were trained under a completely different version of the code. The loss of those models were not of the order of O(0.01) but O(0.001).

![Image](https://github.com/user-attachments/assets/860df36a-f719-447e-8677-79e81e81a2fe)

**Comment by clessig** [comment]
Can you run evaluation on the best model and add the global RMSE values for the standard headline scores.

**Comment by SavvasMel** [comment]
> [@SavvasMel](https://github.com/SavvasMel) : the models with O(0.001) loss were just-pass through without masking, right?

It was definitely without masking, it was one of the pretrained models you uploaded somewhere and you told me to use for forecasting.

**Comment by SavvasMel** [comment]
> Can you run evaluation on the best model and add the global RMSE values for the standard headline scores.

Please go check here https://github.com/ecmwf/WeatherGenerator/issues/242#issuecomment-2910492220

**Comment by SavvasMel** [comment]
assign me
